### PR TITLE
Remove references to server side routing

### PR DESCRIPTION
**Pull request by marcushellberg**
Updated the Hilla routing doc to remove mentions of server-side routes, and to add an example of accessing route parameters.

### CODE REVIEW

File: articles/routing/defining.asciidoc [hunk 1]
```diff
@@ -6,100 +6,67 @@ layout: page
 
 = Defining Routes
 
-TypeScript routes, typically for views or sub-views, are defined in a `Router` instance in the `index.ts` script.
-If you also have server-side views defined in Java, you need to configure the router to fall back to those.
+Routes for views or sub-views, are defined in the `routes.ts` file.
```
Comment (dougjewsbury) [inline articles/routing/defining.asciidoc:9]:
```suggestion
Routes for views or sub-views are defined in the `routes.ts` file.
```

File: articles/routing/defining.asciidoc [hunk 2]
```diff
@@ -6,100 +6,67 @@ layout: page
...
-
-[source,typescript]
-----
-import { Flow } from '@vaadin/flow-frontend';
-const flow = new Flow({
-  imports: () => import('../target/frontend/generated-flow-imports')
-});
+router.setRoutes(routes); // <2>
 ----
+<1> Create a new `Router` instance that outputs its content into the element with the `outlet` id in `index.html`.
```
Comment (dougjewsbury) [inline articles/routing/defining.asciidoc:70]:
```suggestion
<1> Create a new `Router` instance that outputs its content into the element that has the id `outlet` in `index.html`.
```

File: articles/routing/defining.asciidoc [hunk 3]
```diff
@@ -6,100 +6,67 @@ layout: page
...
+This way, the views and their dependencies will only get downloaded if the user navigates to the view. 
+<4> Optional: define a main layout for things like header, footer, and navigation. 
+Include the views as children.
 
-== Pure Client Side Routing
+== Initializing the Router
 
-In a pure TypeScript application with no server-side Java routes, you can configure routing as follows:
+The router is initialized in `index.ts` using the routes defined in `routes.ts`. 
+The router in
```
Comment (dougjewsbury) [inline articles/routing/defining.asciidoc:58]: Incomplete sentence?

Comment (marcushellberg) [reply articles/routing/defining.asciidoc:58]: Thanks

**Review by dougjewsbury — COMMENTED** [review]
AsciiDoc looks good.

**Comment by jouni** [comment]
I’m hesitant in reviewing the content, I’m not that familiar with Hilla routing. @platosha, could you take care of this review?

**Review by platosha — APPROVED** [review]

**Pull request merged**